joplin/templates/wagtailadmin/shared/header.html

```diff
@@ -0,0 +1,51 @@
+{% load i18n %}
+{% comment %}
+
+    Variables accepted by this template:
+
+    title
+    subtitle
+
+    search_url - if present, display a search box. This is a URL route name (taking no parameters) to be used as the action for that search box
+    query_parameters - a query string (without the '?') to be placed after the search URL
+
+    icon - name of an icon to place against the title
+
+    tabbed - if true, add the classname 'tab-merged'
+    merged - if true, add the classname 'merged'
+
+    add_link - if present, display an 'add' button. This is a URL route name (taking no parameters) to be used as the link URL for the button
+    add_text - text for the 'add' button
+{% endcomment %}
+<header class="{% if merged %}merged{% endif %} {% if tabbed %}tab-merged{% endif %} {% if search_form %}hasform{% endif %}" role="banner">
```

Comment:
I see this is where hasform gets set, just curious as to when/why we're using it to set display:none

Reply:
This is a vanilla wagtail thing (all we're doing is hiding the part thats commented), but I believe the submit is hidden because it autocomplets the search as you type, so having the submit button there would be kinda redundant, but the submit is still there for the form to work.  

Not 100% sure if we're talking about the same part though, happy to chat about it.